File: python/nestBattery/nestBattery.py
from novum_api_client.api_type import (
    TInsights,
    TBatteryReading,
    TTreeProperties,
    TBatteryRequired,
)
from novum_api_client.client import NovumAPIClient

api = NovumAPIClient()

try:
    login = api.login("YOUR_EMAIL", "YOUR_PASSWORD")
    # In order to get the "id" of the require battery one can do that copying the id from the Service Center or querying your battery and get its id
    parent_battery = api.get_battery_by_id("YOUR_BATTERY_ID")
    enable_parent = TBatteryReading(
        name=parent_battery.name,
        battery_type=parent_battery.battery_type,
        id=parent_battery.id,
        insights=TInsights(enabled=True),
    )
    update = api.update_battery(enable_parent)
    # after the parent battery is enable to get children, one can create batteries like the following:
    children_battery = api.create_battery(
        TBatteryRequired(
            name="First Cell",
            tree=TTreeProperties(parent=parent_battery.id),
            battery_type=parent_battery.battery_type,
    )
    )

except:
    raise Exception("It was not possible to fetch your battery.")

File: typescript/nestBattery/nestBattery.ts
import { PublicAPIClient as APIClient } from "@novum-batteries/cloud-api-client";

const apiClient = new APIClient();

(async () => {
  try {
    await apiClient.login("E-MAIL", "PASSWORD");
    // In order to update information of a battery you need to define Partial<TBattery> object where id is mandatory
    const parentBattery = await apiClient.getBatteryById("BATTERY_ID");
    const NEST_BATTERY = {
      id: parentBattery.id,
      "tree.enabled": true,
      "insights.enabled": true,
    };
    const update = await apiClient.updateBattery(NEST_BATTERY);
    // after the parent battery is enable to get childrens, one can create batteries like the following:
    const childrenBattery = await apiClient.createBattery({
      name: "First children",
      "tree.parent": parentBattery.id,
      battery_type: parentBattery.battery_type,
    });
    console.log(childrenBattery);
  } catch (e) {
    console.error(e);
  }
})();
